engine/flink/avro-util/src/main/scala/pl/touk/nussknacker/engine/avro/encode/BestEffortAvroEncoder.scala

```diff
@@ -136,8 +136,8 @@ class BestEffortAvroEncoder(avroSchemaEvolution: AvroSchemaEvolution) {
     fields.asScala.map {
       case (fieldName, value) =>
         val field = schema.getField(fieldName)
-        if (field == null) {
-          error(s"Not expected field with name: $fieldName for schema: $schema")
+        if (field == null && !validationMode.acceptRedundant) {
+          error(s"Not expected field with name: $fieldName for schema: $schema and policy $validationMode does not allow redundant")
```

Comment:
How it will work? Can you add some test for that? In case of `field == null && validationMode.acceptRedundant` I see NPE in here: `field.schema()` :)

Reply:
done...